base/file.jl

```diff
@@ -467,23 +469,29 @@ function walkdir(root; topdown=true, follow_symlinks=false, onerror=throw)
         end
     end
 
-    function _it()
+    function _it(chnl)
+        close(chnl, current_task())
         if topdown
-            produce(root, dirs, files)
+            put!(chnl, (root, dirs, files))
+            yield()
```

Comment:
`put!` only had a `notify` and not an explicit yield. Have added a `yield` to `put!` now.